server/server_test.go

```diff
@@ -102,30 +93,45 @@ func (s *testLeaderServerSuite) TearDownSuite(c *C) {
 	s.client.Close()
 }
 
+func (s *testLeaderServerSuite) setUpClient(c *C) {
+	endpoints := make([]string, 0, 3)
+
+	for _, svr := range s.svrs {
+		endpoints = append(endpoints, svr.GetEndpoints()...)
+	}
+
+	var err error
+	s.client, err = clientv3.New(clientv3.Config{
+		Endpoints:   endpoints,
+		DialTimeout: 3 * time.Second,
+	})
+	c.Assert(err, IsNil)
+}
+
 func (s *testLeaderServerSuite) TestLeader(c *C) {
 	for _, svr := range s.svrs {
 		go svr.Run()
 	}
 
-	for i := 0; i < 100 && len(s.svrs) > 0; i++ {
-		leader, err := getLeader(s.client, s.leaderPath)
-		c.Assert(err, IsNil)
+	leader1 := mustGetLeader(c, s.client, s.leaderPath)
+	svr, ok := s.svrs[leader1.GetAddr()]
+	c.Assert(ok, IsTrue)
+	svr.Close()
+	delete(s.svrs, leader1.GetAddr())
 
-		if leader == nil {
-			time.Sleep(500 * time.Millisecond)
-			continue
-		}
+	// now, another two servers must select a leader
+	s.setUpClient(c)
```

Comment:
When I wrote these codes, etcd has a bug that client still uses closed server, but it is fixed later. 
I will update etcd vendor later. As you know, updating vendor is a terrible thing, I will do it next PR. So let's keep it here now.